src/main/java/it/gov/pagopa/service/PortalConsentServiceImpl.java

```diff
@@ -1,14 +1,63 @@
 package it.gov.pagopa.service;
 
+import it.gov.pagopa.connector.onetrust.OneTrustRestService;
+import it.gov.pagopa.dto.PortalConsentDTO;
+import it.gov.pagopa.dto.mapper.PrivacyNotices2PortalConsentDTOMapper;
+import it.gov.pagopa.dto.onetrust.PrivacyNoticesDTO;
+import it.gov.pagopa.model.PortalConsent;
 import it.gov.pagopa.repository.PortalConsentRepository;
-import it.gov.pagopa.repository.RolePermissionRepository;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
 
+import java.time.LocalDateTime;
+import java.util.Optional;
+
 @Service
 public class PortalConsentServiceImpl implements PortalConsentService{
+
     @Autowired
     PortalConsentRepository portalConsentRepository;
 
+    @Autowired
+    OneTrustRestService oneTrustRestService;
+
+    @Autowired
+    PrivacyNotices2PortalConsentDTOMapper privacyNotices2PortalConsentDTOMapper;
+
+    @Value("${onetrust.privacy-notices.tos.id}")
+    private String tosId;
+
+    @Override
+    public PortalConsentDTO get(String userId) {
+
+        Optional<PortalConsent> optional = portalConsentRepository.findById(userId);
```

Comment:
let the `findById` return the real object and use the `orElseThrow` method to catch Exception on no User found to raise an Http NotFound controlled by `@ControllerAdvice`